tools/engine_tool/lib/src/build_plan.dart

```diff
@@ -0,0 +1,279 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:collection/collection.dart';
+import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:meta/meta.dart';
+
+import 'build_utils.dart';
+import 'environment.dart';
+import 'logger.dart';
+
+const _flagConfig = 'config';
```

Comment:
Why are we pulling flags out of flags.dart?

Reply:
They are no longer visible or relevant to the commands (they are implementation details).

Reply:
That is, the reason we had them in a common place no longer applies.